azure-communication-ui/calling/src/main/java/com/azure/android/communication/ui/calling/redux/middleware/handler/CallingMiddlewareActionHandler.kt

```diff
@@ -358,21 +383,35 @@ internal class CallingMiddlewareActionHandlerImpl(
                     tryCameraOn(store)
                 }
 
+                if (store.getCurrentState().callState.operationStatus == OperationStatus.SKIP_SETUP_SCREEN &&
+                    callInfoModel.callingStatus == CallingStatus.CONNECTED
+                ) {
+                    tryCameraOn(store)
+                }
+
                 callInfoModel.callStateError?.let {
                     val action = ErrorAction.CallStateErrorOccurred(it)
                     store.dispatch(action)
                     if (it.callCompositeEventCode == CallCompositeEventCode.CALL_EVICTED ||
                         it.callCompositeEventCode == CallCompositeEventCode.CALL_DECLINED
                     ) {
-                        store.dispatch(NavigationAction.SetupLaunched())
+                        if (store.getCurrentState().callState.operationStatus == OperationStatus.SKIP_SETUP_SCREEN) {
+                            store.dispatch(NavigationAction.Exit())
+                        } else {
+                            store.dispatch(NavigationAction.SetupLaunched())
+                        }
                     } else if (it.errorCode == ErrorCode.CALL_END_FAILED ||
                         it.errorCode == ErrorCode.CALL_JOIN_FAILED
                     ) {
                         store.dispatch(CallingAction.IsTranscribingUpdated(false))
                         store.dispatch(CallingAction.IsRecordingUpdated(false))
                         store.dispatch(ParticipantAction.ListUpdated(HashMap()))
                         store.dispatch(CallingAction.StateUpdated(CallingStatus.NONE))
-                        store.dispatch(NavigationAction.SetupLaunched())
+                        if (store.getCurrentState().callState.operationStatus == OperationStatus.SKIP_SETUP_SCREEN) {
+                            store.dispatch(NavigationAction.Exit())
```

Comment:
Did we test this for memory leaks when exiting like this? Not suggesting it will cause any, just in case?

Reply:
Good call. I checked it. I don't see memory leak. The reason being, it happens when call evicted or network fails, this comes as sdk event so the call sdk resource hold up is already cleaned.
With no skip setup -> we used to navigate to setup screen
With setup screen -> we would just navigate to exit.